Kong Manager landing page

cloudjumpercat opened the pull request:
## Description

Fixes #261 

## Preview Links

- https://deploy-preview-643--kongdeveloper.netlify.app/gateway/kong-manager/
- (added customization section to this page) https://deploy-preview-643--kongdeveloper.netlify.app/gateway/kong-manager-configuration/#customize-kong-manager

Stub page issues created:
- https://github.com/Kong/developer.konghq.com/issues/666

## Notes for reviewers

I pulled content from these two source files ([here](https://docs.konghq.com/gateway/latest/kong-manager/auth/) and [here](https://docs.konghq.com/gateway/latest/kong-manager/)), but scrapped a lot of the content on the pages because 
- It was taking users on a tour of the UI, which I thought wasn't as useful as a list of what could be managed in KM
- The authentication details should be on the linked how to pages, so I omitted them from the landing page


## Checklist 

- [x] Every page is page one.
- [ ] Tested how-to docs. If not, note why here. *N/A: landing page*
- [x] All pages contain metadata.
- [x] Updated [sources.yaml](https://github.com/Kong/developer.konghq.com/blob/main/tools/track-docs-changes/config/sources.yml). For more info, review [track docs changes](https://github.com/Kong/developer.konghq.com/tree/main/tools/track-docs-changes)
- [x] Any new docs link to existing docs.
- [x] All autogenerated instructions render correctly (API, decK, Konnect, Kong Manager). 
- [x] Style guide (capitalized gateway entities, placeholder URLs) implemented correctly.

lena-larionova commented on app/_landing_pages/gateway/kong-manager.yaml:
```diff
@@ -0,0 +1,147 @@
+                    - Create and manage Routes and Gateway Services
+                    - Activate or deactivate plugins
+                    - Manage Certificates
+                    - Manage users and roles for {{site.base_gateway}}
```
```suggestion
                    - Manage RBAC for {{site.base_gateway}}:  users, admins, roles, permissions, and teams
```

lena-larionova commented on app/_landing_pages/gateway/kong-manager.yaml:
```diff
@@ -0,0 +1,147 @@
+      - blocks:
+          - type: image
+            config:
+              url: /assets/images/gateway/kong-manager-ui.png
```
I wonder if we can resize the screen to a narrower width. Right now it's really small and hard to see the screenshot.

cloudjumpercat replied:
Sure! I've fixed this in my latest commit

lena-larionova commented on app/_landing_pages/gateway/kong-manager.yaml:
```diff
@@ -0,0 +1,147 @@
+                  If you're running {{site.base_gateway}} on-prem with a database (either in traditional
+                  or hybrid mode), you can [enable Kong Manager](/gateway/enable-kong-manager/). If you run the [{{site.base_gateway}} quickstart script](https://get.konghq.com/quickstart), Kong Manager is automatically enabled.
+
+                  If you want to enable Kong Manager for multiple domains in {{site.base_gateway}} 3.9.x or later, see [Multiple domains](/gateway/enable-kong-manager/#multiple-domains).
```
```suggestion
                  If you want to enable Kong Manager for multiple domains in {{site.base_gateway}} 3.9.x or later, see [multiple domains](/gateway/enable-kong-manager/#multiple-domains).
```

lena-larionova commented on app/_landing_pages/gateway/kong-manager.yaml:
```diff
@@ -0,0 +1,147 @@
+
+                  If you want to enable Kong Manager for multiple domains in {{site.base_gateway}} 3.9.x or later, see [Multiple domains](/gateway/enable-kong-manager/#multiple-domains).
+                  
+                  To access Kong Manager, go to the following URL: [http://localhost:8002](http://localhost:8002)
```
This is only true if you're running locally and you probably used the quickstart + default values. Maybe rephrase this to something like "Access Kong Manager at its default URL: "

lena-larionova commented on app/_landing_pages/gateway/kong-manager.yaml:
```diff
@@ -0,0 +1,147 @@
+            blocks:
+              - type: text
+                text: |
+                  {{site.base_gateway}} comes packaged with authentication plugins that can be used to secure Kong Manager.
```
```suggestion
                  Kong Manager comes packaged with several authentication methods.
```
While they are sort of using the Kong plugins in the backend, you don't configure Kong plugins to secure Kong Manager. It's a weird tangential config method that, to the user, is only applicable in the Kong Manager context.

lena-larionova commented on app/_landing_pages/gateway/kong-manager.yaml:
```diff
@@ -0,0 +1,147 @@
+          config:
+            title: Enable OIDC for Kong Manager
+            description: |
+              Kong Gateway offers the ability to bind authentication for Kong Manager admins to an organization’s OpenID Connect Identity Provider using the OpenID Connect Plugin.
```
```suggestion
              {{site.base_gateway}} offers the ability to bind authentication for Kong Manager admins to an organization’s OpenID Connect Identity Provider.
```

lena-larionova commented on app/_landing_pages/gateway/kong-manager.yaml:
```diff
@@ -0,0 +1,147 @@
+          config:
+            title: Enable Basic Auth for Kong Manager
+            description: |
+              Configure Basic Auth authentication with Kong Manager.
```
```suggestion
              Configure basic authentication for Kong Manager.
```

lena-larionova commented on app/_landing_pages/gateway/kong-manager.yaml:
```diff
@@ -0,0 +1,147 @@
+        - type: structured_text
+          config:
+            header:
+              text: "Access control with roles and Workspaces"
```
```suggestion
              text: "Access control with Roles and Workspaces"
```

lena-larionova reviewed (changes requested):
* Adjust the screenshot to make it slightly clearer
* Don't mention plugins when talking about KM auth, we don't need to expose that to users
* I think we're missing something about sessions?
* We should add how-tos and references at the bottom, which will catch things like sending email, network config, and any specific auth how-tos.
* Would be useful to somehow link to the Kong Manager section of the kong.conf ref: https://docs.konghq.com/gateway/latest/reference/configuration/#kong-manager-section
* Should we add a customization reference (maybe post-GA, doesn't exist now)? There's a bunch of config parameters in kong.conf that let you adjust the header and footer. Or just add a link to it somewhere, somehow.

lena-larionova commented on app/_landing_pages/gateway/kong-manager.yaml:
```diff
@@ -0,0 +1,147 @@
+                    - Manage Certificates
+                    - Manage users and roles for {{site.base_gateway}}
+                    - Centrally store and easily access Key Sets and Keys
+                    - Manage Vaults
```
```suggestion
                    - Manage Vaults
                    - Customize the header and footer for your own organization
```

cloudjumpercat commented:
@lena-larionova Thanks for the review! A few things I changed that I wanted to note:

- I added the [References section to the landing page](https://deploy-preview-643--kongdeveloper.netlify.app/gateway/kong-manager/#references-tools). Looks like it's bugged and is pulling different versions of the same page. I think this is okay to not fix for now since reference page versions will be going away.
- Per your comment about the customization settings, I just quickly [threw a section on the "Kong Manager Configuration"](https://deploy-preview-643--kongdeveloper.netlify.app/gateway/kong-manager-configuration/#customize-kong-manager) reference page. Let me know what you think about this and the location!

lena-larionova reviewed (approved):
One minor adjustment, otherwise lgtm!

> I added the [References section to the landing page](https://deploy-preview-643--kongdeveloper.netlify.app/gateway/kong-manager/#references-tools). Looks like it's bugged and is pulling different versions of the same page. I think this is okay to not fix for now since reference page versions will be going away.

Yeah, we can ignore this, it'll get fixed.

> Per your comment about the customization settings, I just quickly [threw a section on the "Kong Manager Configuration"](https://deploy-preview-643--kongdeveloper.netlify.app/gateway/kong-manager-configuration/#customize-kong-manager) reference page. Let me know what you think about this and the location!

I think this works!
